### PR TITLE
No issue: make StrictMode suppression count thread safe.

### PR DESCRIPTION
**Pull request by mcomella**
Missed this one.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/StrictModeManager.kt [hunk 1]
```diff
@@ -120,11 +123,11 @@ class StrictModeManager(
             // This can overflow and crash. However, it's unlikely we'll suppress StrictMode 9
             // quintillion times in a build config where StrictMode is enabled so we don't handle it
             // because it'd increase complexity.
-            suppressionCount += 1
+            val suppressionCount = suppressionCount.incrementAndGet()
 
             // We log so that devs are more likely to notice that we're suppressing StrictMode violations.
             // We add profiler markers so that the perf team can easily identify IO locations in profiles.
-            logger.warn("StrictMode violation suppressed: #$suppressionCount")
+            logger.warn("StrictMode violation suppressed: #${suppressionCount}")
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/StrictModeManager.kt:130]: nit (while we're changing this): violation(s)?

File: app/src/main/java/org/mozilla/fenix/StrictModeManager.kt [hunk 2]
```diff
@@ -49,9 +50,11 @@ class StrictModeManager(
      * - a lint check: to ensure this value gets used instead of functions that work around it
      * - code owners: to prevent modifications to these above items without perf knowing
      * to make suppressions a more deliberate act.
+     *
+     * This is an Atomic* so it can be incremented from any thread.
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/StrictModeManager.kt:54]: nit: *safely* incremented ... the non-atomic one we can increment from any thread as well :)

**Review by csadilek — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/StrictModeManager.kt (hunk 1 above)
Comment (mcomella) [reply app/src/main/java/org/mozilla/fenix/StrictModeManager.kt:130]: The way I intended this is, "A StrictMode violation was suppressed RIGHT NOW. It's the X instance of it." I feeling like changing it to "violations" is less clear: did the violation happen just now? Did multiple violations happen at once? I'd rather clarify "this is the # X instance of it" but I'm not sure how to do that

**Comment by mcomella** [comment]
ktlint failure is: 

> [task 2020-11-02T18:21:38.813Z] /builds/worker/checkouts/src/app/src/main/java/org/mozilla/fenix/StrictModeManager.kt:130:28: Redundant curly braces

**Pull request merged**